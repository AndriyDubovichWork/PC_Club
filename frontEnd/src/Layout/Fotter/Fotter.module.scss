.Fotter {
  width: 80%;
  margin: 0 auto;
  padding: 5vh 0;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  grid-column-gap: 1vw;
  grid-row-gap: 5vh;
  .SubTitle {
    margin: 2vh 0;
  }
  .HighLighted {
    display: flex;
    align-items: center;

    cursor: pointer;

    color: #000;

    margin: 1vh 0;
  }
  .HighLighted:hover {
    color: #ff4d30;
  }

  .Input {
    border: 0;
  }
}
