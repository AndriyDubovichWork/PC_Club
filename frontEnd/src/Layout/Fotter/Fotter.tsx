import React, { useContext } from 'react';
import style from './Fotter.module.scss';
import LocalPhoneIcon from '@mui/icons-material/LocalPhone';
import EmailIcon from '@mui/icons-material/Email';
import TextField from '@mui/material/TextField';
import Button from '@mui/material/Button';
import { RefContext } from '../../Context/RefContext';

const Fotter = () => {
  const { HeaderRef, scrollToElement } = useContext(RefContext);

  const companyies = ['New York', 'Careers', 'Mobile', 'Blog', 'How we work'];
  const contents = [
    <div>
      <h1>PC CLUB</h1>
      <p className={style.SubTitle}>
        We offers a big range of service for all your needs. We have the perfect
        PC to meet your needs
      </p>
      <div className={style.HighLighted}>
        <LocalPhoneIcon />
        (123) -456-789
      </div>
      <div className={style.HighLighted}>
        <EmailIcon />
        randomemail@gmail.com
      </div>
    </div>,
    <div>
      <h1>COMPANY</h1>
      {companyies.map((company) => {
        return (
          <p
            key={company}
            className={style.HighLighted}
            onClick={() => {
              scrollToElement(HeaderRef);
            }}
          >
            {company}
          </p>
        );
      })}
    </div>,
    <div>
      <h1>WORKING HOURS</h1>
      <p className={style.HighLighted}>Mon - Fri: 9:00AM - 9:00PM</p>
      <p className={style.HighLighted}>Sat: 9:00AM - 19:00PM</p>
      <p className={style.HighLighted}>Sun: Closed</p>
    </div>,
    <div>
      <h1>SUBSCRIPTION</h1>
      <p>Subscribe your Email address for latest news & updates.</p>
      <TextField
        variant='filled'
        className={style.Input}
        placeholder='Enter Email Address'
        type='email'
        name='email'
        sx={{ width: '100%', margin: '2vh 0' }}
      />
      <Button
        variant='contained'
        sx={{
          width: '100%',
          backgroundColor: '#ff4d30',
          '&:hover': {
            backgroundColor: '#ff4d30',
          },
        }}
      >
        Submit
      </Button>
    </div>,
  ];

  return (
    <>
      <footer className={style.Fotter}>
        {contents.map((content, id) => {
          return <div key={id}>{content}</div>;
        })}
      </footer>
    </>
  );
};

export default Fotter;
