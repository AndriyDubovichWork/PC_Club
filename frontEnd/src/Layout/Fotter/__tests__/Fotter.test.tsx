import { render, screen, cleanup } from '@testing-library/react';
import Fotter from '../Fotter';
import renderer from 'react-test-renderer';
afterEach(() => {
  cleanup();
});

test('Fotter match snapshot', () => {
  const tree = renderer.create(<Fotter />).toJSON();
  expect(tree).toMatchSnapshot();
});
