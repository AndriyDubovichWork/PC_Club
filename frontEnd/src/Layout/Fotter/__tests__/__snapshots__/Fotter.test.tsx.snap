// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fotter match snapshot 1`] = `
<footer
  className="Fotter"
>
  <div>
    <div>
      <h1>
        PC CLUB
      </h1>
      <p
        className="SubTitle"
      >
        We offers a big range of service for all your needs. We have the perfect PC to meet your needs
      </p>
      <div
        className="HighLighted"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LocalPhoneIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
          />
        </svg>
        (123) -456-789
      </div>
      <div
        className="HighLighted"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="EmailIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
          />
        </svg>
        randomemail@gmail.com
      </div>
    </div>
  </div>
  <div>
    <div>
      <h1>
        COMPANY
      </h1>
      <p
        className="HighLighted"
        onClick={[Function]}
      >
        New York
      </p>
      <p
        className="HighLighted"
        onClick={[Function]}
      >
        Careers
      </p>
      <p
        className="HighLighted"
        onClick={[Function]}
      >
        Mobile
      </p>
      <p
        className="HighLighted"
        onClick={[Function]}
      >
        Blog
      </p>
      <p
        className="HighLighted"
        onClick={[Function]}
      >
        How we work
      </p>
    </div>
  </div>
  <div>
    <div>
      <h1>
        WORKING HOURS
      </h1>
      <p
        className="HighLighted"
      >
        Mon - Fri: 9:00AM - 9:00PM
      </p>
      <p
        className="HighLighted"
      >
        Sat: 9:00AM - 19:00PM
      </p>
      <p
        className="HighLighted"
      >
        Sun: Closed
      </p>
    </div>
  </div>
  <div>
    <div>
      <h1>
        SUBSCRIPTION
      </h1>
      <p>
        Subscribe your Email address for latest news & updates.
      </p>
      <div
        className="MuiFormControl-root MuiTextField-root Input css-1p1uke1-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1ff8729-MuiInputBase-root-MuiFilledInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
            disabled={false}
            id=":r0:"
            name="email"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter Email Address"
            required={false}
            type="email"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1cvwac4-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
</footer>
`;
